CONTRIBUTING.md

```diff
@@ -1,70 +1,130 @@
+project codebases, issue trackers, chat-rooms, and mailing lists.
 
-1. Run `./bin/setup`.
+[code of conduct]: https://thoughtbot.com/open-source-code-of-conduct
+
+## Getting Started
+
+Administrate is a [Rails Engine][], but ships with everything needed to
+contribute and test new changes.
+
+To maintain compatibility with multiple dependency versions, we use
+[Appraisal][].
+
+[Rails Engine]: https://guides.rubyonrails.org/engines.html
+[Appraisal]: https://github.com/thoughtbot/appraisal
+
+### Opening a PR
+
+1. Fork the repo,
+2. Run `./bin/setup` to install the base dependencies and setup a local 
+   database,
+3. Run the test suite: `bundle exec rspec && bundle exec appraisal rspec`,
+4. Make your changes,
+5. Push your fork and open a pull request.
+
+A good PR will solve the smallest problem it possibly can, have good test
+coverage and (where necessary) have internationalisation support.
+
+### Running the application locally
 
-1. Run `appraisal install`
+Administrate's demo application can be run like any Rails application:
 
-1. Run the tests. We only take pull requests with passing tests, and it's great
-   to know that you have a clean slate: `bundle exec rake && bundle exec appraisal rake`
+```sh
+bundle exec rails s
+```
 
-1. Add a test for your change. Only refactoring and documentation changes
-   require no new tests. If you are adding functionality or fixing a bug,
-   we need a test!
+This will start the application defined in `spec/example_app`.
 
-1. Make the test pass.
+## Repository Structure
 
-1. Write a [good commit message][commit].
+* The gem's source code lives in the `app` and `lib` subdirectories.
+* The demo app is nested within `spec/example_app`.
+* The guides as seen at
+  [https://administrate-demo.herokuapp.com][docs] live as
+  markdown files in the `docs` subdirectory.
 
-1. Push to your fork and submit a pull request.
+Rails configuration files have been changed
+to recognize the app in the new location,
+so running the server or deploying to Heroku works normally.
 
-Others will give constructive feedback.
-This is a time for discussion and improvements,
-and making the necessary changes will be required before we can
-merge the contribution.
+With this structure, developing a typical feature looks like:
 
-## Start Application in Development
+* Add tests in `spec/`
+* Implement a feature in `administrate/`
+* Exercise the feature using the demo rails app (`spec/example_app/app/`)
 
-Configure your local environment with `./bin/setup`.
-After that start the application with the `foreman start` command.
+[demo]: https://administrate-prototype.herokuapp.com/admin
+[docs]: https://administrate-prototype.herokuapp.com
 
-## Performance Improvements
+## Front-end Architecture
 
-Improving our users' experience should be the primary goal of any optimization.
+This project uses:
 
-If you contribute a performance improvement,
-you must submit performance profiles
```

Comment:
Apparently this was added because [people were trying to micro-optimize in this codebase](https://github.com/thoughtbot/administrate/pull/293). Is it worth keeping something like this around?

Reply:
This isn't something I've seen for a long time, so I think it's okay that it's no longer there.

Reply:
Sounds good. We can bring it back if it pops up again.